### DISCUSSION TITLE
Ability to specify a custom executable for IntelliJ plugin for running format

### DISCUSSION THREAD

**Discussion by Joshuabaker2** [biomejs/biome#662] [Ideas]
Would there be any interest in the ability to specify a custom executable path for the IntelliJ plugin? By default, it uses your built-in interpreter (which is a very sane default). But, it might be nice to be able to specify a different interpreter, for example, Bun, to run the formatting.

This might be a niche use-case, my interpreter is node and my package manager is yarn, which is required for my project to work correctly in the IDE, but starting biome through yarn/node is significantly slower than starting biome through bun.

```
> time yarn biome format .
Compared 2615 file(s) in 237ms
yarn biome format .  2.11s user 0.35s system 267% cpu 0.920 total

> time bun run biome format .
Compared 2615 file(s) in 229ms
bun run biome format .  1.30s user 0.24s system 535% cpu 0.289 total
```

The major motivator for me having moved to Biome from Eslint/Prettier was the massive speedup, so I feel like this could fit into the philosophy of this project. I hackily modified a local version of the plugin to work for me using bun instead and it is noticeably faster. If others might get value from it, I'd be happy to submit a PR for it.

**Comment by ematipico** [comment]
I am not against the proposal. I wonder if we allow such an option somehow, what would happen if the interpreter causes an error when running the biome binary? Can the user see the error message? And if the user decides to file an issue, how should behave? Is it the plugin's fault or the interpreter's fault?  

Also, is it needed? IntelliJ just released support for formatting: https://youtrack.jetbrains.com/issue/IDEA-334150 

This means we will change how we format the code later
